integ-search-service/src/main/java/org/exoplatform/commons/search/service/UnifiedSearchService.java

```
@@ -295,13 +296,16 @@ public Response REST_getSearchSetting() {
   */
   @POST
   @Path("/setting")
+  @RolesAllowed("administrators")
   public Response REST_setSearchSetting(@FormParam("resultsPerPage") long resultsPerPage, @FormParam("searchTypes") String searchTypes, @FormParam("searchCurrentSiteOnly") boolean searchCurrentSiteOnly, @FormParam("hideSearchForm") boolean hideSearchForm, @FormParam("hideFacetsFilter") boolean hideFacetsFilter) {
-    settingService.set(Context.GLOBAL, Scope.WINDOWS, "searchResult_resultsPerPage", new SettingValue<Long>(resultsPerPage));    
+    if (resultsPerPage > 0) {
+      settingService.set(Context.GLOBAL, Scope.WINDOWS, "searchResult_resultsPerPage", new SettingValue<Long>(resultsPerPage));
+    }
```

I think this change is not related to the current issue. Moreover, if the resultsPerPage is not an acceptable value (not > 0), we should return an error instead of silently ignoring it IMO.
So I think we should remove this change from this PR. WDYT ?